segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/StorageLoader.java

```diff
@@ -53,7 +56,11 @@ public StorageFactory load(ConfigSetup setup,
                     if (storageExtraConfig.isStorageNoOpMode()) {
                         //The specified storage implementation is in No-Op mode.
                         log.warn("{} IS IN NO-OP MODE: DATA LOSS WILL HAPPEN! MAKE SURE IT IS BY FULL INTENTION FOR TESTING PURPOSE!", storageImplementation);
-                        return new NoOpStorageFactory(storageExtraConfig, executor, factory, null);
+                        if (factory instanceof SimpleStorageFactory) {
```

Comment:
There is a build error with spotbugs.
Please use temp variable to reduce number of return statements